### PR TITLE
Fixing typos, grammatical and spelling mistakes in Refactorization Overview document

### PR DESCRIPTION
**Pull request by abhinavarora**

### CODE REVIEW

File: doc/design/refactorization.md [hunk 1]
```diff
@@ -177,34 +178,34 @@ REGISTER_OP(op_type, op_class, op_maker_class, grad_op_type, grad_op_class)
...
-1. Write Op class, as well as its gradient Op class if there is.
-2. Write Op maker class. In the constructor, describe its inputs, outputs, and attributes.
-3. Invoke macro `REGISTER_OP`. The macro will
-	1. call maker class to complete `proto` and `checker`
-	2. with the completed `proto` and `checker`, build a new key-value pair in the `OpInfoMap`
+# Registration Process
+1. Write an Op class and its gradient Op class, if required.
+2. Write an Op maker class. In the constructor of this class, describe the inputs, outputs and attributes of the operator.
+3. Invoke the macro `REGISTER_OP`. This macro will
+	1. Call maker class to complete the `proto` and the `checker`
```
Comment (NorthStar) [inline doc/design/refactorization.md]: Drop the articles here.

File: doc/design/refactorization.md [hunk 2]
```diff
@@ -177,34 +178,34 @@ REGISTER_OP(op_type, op_class, op_maker_class, grad_op_type, grad_op_class)
...
+	2. Using the completed `proto` and `checker`, it will add a new key-value pair to the `OpInfoMap`
```
Comment (NorthStar) [inline doc/design/refactorization.md]: Add a new key-value pair to `OpInfoMap` using the completed `proto` and `checker`.

File: doc/design/refactorization.md [hunk 3]
```diff
@@ -177,34 +178,34 @@ REGISTER_OP(op_type, op_class, op_maker_class, grad_op_type, grad_op_class)
...
 
-4. Invoke `USE` macro in where the Op is used to make sure it is linked.
+4. Invoke the `USE` macro in which the Op is used, to make sure that it is linked.
```
Comment (NorthStar) [inline doc/design/refactorization.md]: Drop the comma here.

Comment (NorthStar) [reply doc/design/refactorization.md]: `Invoke the `USE` macro where an Op is used to make sure that it is linked.`

File: doc/design/refactorization.md [hunk 4]
```diff
@@ -177,34 +178,34 @@ REGISTER_OP(op_type, op_class, op_maker_class, grad_op_type, grad_op_class)
...
 ---
 # Backward Module (2/2)
 ### Build Backward Network
-- **Input** graph of forwarding operators
-- **Output** graph of backward operators
-- **corner case in construction**
-	- shared variable => insert `Add` operator
-	- no gradient => insert `fill_zero_grad` operator
-	- recursive netOp => call `Backward` recursively
+- **Input**: graph of forwarding operators
```
Comment (NorthStar) [inline doc/design/refactorization.md]: _a_ graph

File: doc/design/refactorization.md [hunk 5]
```diff
@@ -213,41 +214,41 @@ make sure the registration process is executed and linked.
...
-	* All operators on `Tensor` is written in `Operator` or global functions.
-	* variable length Tensor design [LoDTensor](https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/framework/lod_tensor.md)
-* `Variable` is the inputs and outputs of an operator. Not just `Tensor`.
-	* step_scopes in RNN is a variable and not a tensor.
-* `Scope` is where variables store at.
-	* map<string/*var name */, Variable>
-	* `Scope` has a hierarchical structure. The local scope can get variable from its parent scope.
+	* All operations on `Tensor` are written in `Operator` or global functions.
+	* Variable length Tensor design [LoDTensor](https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/framework/lod_tensor.md)
+* `Variable` instances are the inputs and the outputs of an operator. Not just `Tensor`.
```
Comment (NorthStar) [inline doc/design/refactorization.md]: , not just `Tensor`.

File: doc/design/refactorization.md [hunk 6]
```diff
@@ -1,71 +1,72 @@
...
+1. Users write Python programs to describe the graphs and run them (locally or remotely).
 
 1. A graph is composed of *variables* and *operators*.
 
-1. The description of graphs must be able to be serialized/deserialized, so it
+1. The description of graphs must be capable of being serialized/deserialized, so that
 
-   1. could to be sent to the cloud for distributed execution, and
-   1. be sent to clients for mobile or enterprise deployment.
+   1. It can to be sent to the cloud for distributed execution, and
```
Comment (pengli09) [inline doc/design/refactorization.md]: s/can to be/can be/g

File: doc/design/refactorization.md [hunk 7]
```diff
@@ -1,71 +1,72 @@
...
-   1. plan for memory-reuse for variables,
-   1. generate backward and optimization part of the Graph.
-   1. possiblly split the graph for distributed training.
+   1. Create `VarDesc` to represent local/intermediate variables,
+   1. Create operators and set attributes,
+   1. Validate attribute values,
+   1. Infer the type and the shape of variables,
+   1. Plan memory-reuse for variables,
+   1. Generate the backward graph
+   1. Optimize the computation graph.
```
Comment (pengli09) [inline doc/design/refactorization.md]: Here "optimization'' refers to the optimization operators related stuff such as Adam, which is also a part of the graph. Therefore, "Optimize the computation graph." is incorrect.

File: doc/design/refactorization.md [hunk 8]
```diff
@@ -105,10 +106,10 @@ Compile Time -> IR -> Runtime
 # Operator
 ![class_diagram](http://api.paddlepaddle.org/graphviz?dot=https://gist.githubusercontent.com/reyoung/53df507f6749762675dff3e7ce53372f/raw/dd598e8f1976f5759f58af5e5ef94738a6b2e661/op.dot)
 
-* `Operator` is the fundamental building block as the user interface.
-    * Operator stores input/output variable name, and attributes.
-    * The `InferShape` interface is used to infer output variable shapes by its input shapes.
-    * Use `Run` to compute `input variables` to `output variables`.
+* `Operator` is the fundamental building block of the user interface.
+    * Operator stores input/output variable names, and attributes.
+    * The `InferShape` interface is used to infer the shape of the output variable shapes based on the shapes of the input variables.
```
Comment (pengli09) [inline doc/design/refactorization.md]: s/the output variable shapes/the output variables/g

File: doc/design/refactorization.md [hunk 9]
```diff
@@ -213,41 +214,41 @@ make sure the registration process is executed and linked.
...
-* `Variable` is the inputs and outputs of an operator. Not just `Tensor`.
-	* step_scopes in RNN is a variable and not a tensor.
-* `Scope` is where variables store at.
-	* map<string/*var name */, Variable>
-	* `Scope` has a hierarchical structure. The local scope can get variable from its parent scope.
+	* All operations on `Tensor` are written in `Operator` or global functions.
+	* Variable length Tensor design [LoDTensor](https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/framework/lod_tensor.md)
+* `Variable` instances are the inputs and the outputs of an operator. Not just `Tensor`.
+	* `step_scopes` in RNN is a variable and not a tensor.
+* `Scope` is where variables are stores.
```
Comment (pengli09) [inline doc/design/refactorization.md]: are stored

File: doc/design/refactorization.md [hunk 10]
```diff
@@ -213,41 +214,41 @@ make sure the registration process is executed and linked.
...
+	* map<string `variable_name`, Variable>
```
Comment (pengli09) [inline doc/design/refactorization.md]: I think `/* var name*/` is better.

File: doc/design/refactorization.md [hunk 11]
```diff
@@ -213,41 +214,41 @@ make sure the registration process is executed and linked.
...
 ---
 # Control the migration quality
-- compare the performance of migrated models with old ones.
-- follow google C style
-- build the automatic workflow of generating Python/C++ documentations
-  - the documentation of layers and ops should be written inside the code
-  - take the documentation quality into account when doing PR
-  - preview the documentations, read and improve them from users' perspective
+- Compare the performance of migrated models with old ones.
+- Follow the google C++ style
```
Comment (pengli09) [inline doc/design/refactorization.md]: s/style/style./g
Note the period.

**Pull request merged**